### ISSUE TITLE
Incorrect display of a turn at a roundabout in the widget

### ISSUE THREAD

**Issue by Zirochkabila** [osmandapp/OsmAnd#16889]
### Description
Since for "normal" crossings the widget for the next turn shows where one should turn (sharply / normally / slightly right, etc.), would be expected that the symbol for passing through the roundabout not only correctly says the number of exit out of it (as it already does) but also reflects the path to take. The path should always come into the roundabout scheme from the bottom (this is the case), but also out at the correct angle and exit number. Currently, the exit number is correct but the angle is sometimes not, and quite gross discrepancies may occur, as shown in the video.
[OSM](https://www.openstreetmap.org/#map=17/33.02536/35.67387)
[OsmAnd](https://osmand.net/map/?start=33.022876%2C35.675794&end=33.024100%2C35.670794&mode=car#17/33.02531/35.67268)
Map here

### Steps to reproduce

- Navigation
- From `33.02288, 35.67579` to `33.02410, 35.67079`

### Actual result

https://user-images.githubusercontent.com/104760013/229815787-83c14a60-773f-4f62-8d5c-4e7529657379.mp4

### Expected result
The path should always come into the roundabout scheme from the bottom (this is the case), but also out at the correct angle and exit number.

### Your Environment
OsmAnd Version: OsmAnd~ 4.5.0#14292mqta, Released: 2023-04-04
Android/iOS version: Android 12
Device model: Poco x3 nfc

**Comment by vshcherb** [comment]
We have no solution except displaying real turn structure in a side map - calculating angle is really faulty operation.
So it's a duplicate

**Comment by Max1234-Ita** [comment]
The strange thing is that for countries where cars drive on the left side of the road (so roundabouts are taken counter-clockwise), it looks to work fine.

Just like this:

<table style="width:100%">
  <tr>
<th><img src="https://github.com/osmandapp/OsmAnd/assets/15315381/a11cb9e1-60d8-42f0-978f-0063001cb4d9"  width="300" height="auto"/> </td>
  </tr>
</table>

More pictures in the English/International Telegram group:

https://t.me/OsmAndMaps/40607?single

(and following posts)

**Comment by pordini** [comment]
The behaviour seems quite inconsistent. These two examples show somewhat opposite behaviour. 

In the first, the widget shows less than 180 degrees while the route is more than that, the second shows way more angle than the route.
<table style="width:100%">
  <tr>
<th><img src="https://github.com/osmandapp/OsmAnd/assets/12774050/59790317-a5f1-40e8-a256-7bfa7f03fad6"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/osmandapp/OsmAnd/assets/12774050/3d10d25d-1b97-4608-bbe5-f03588d3592e"  width="300" height="auto"/> </td>
  </tr>
</table>

**Comment by bobeeeze** [comment]
I'm still having the same problem with version 4.5.10 of Osmand+ on Android. The next turn widget symbol, on the same roundabout, behaves very strangely and contrary to logic depending on the exit I choose.

I've already noticed this problem with version 4.4 and I'm convinced that the symbol was correct before.

I hope this will be corrected in future versions as it's quite confusing when driving.

<table style="width:100%">
<tr>
<td><img src="https://github.com/osmandapp/OsmAnd/assets/59626956/2d892ba1-8bbd-4cbb-9d58-d95459831be1"  width="300" height="auto"/></td>
</tr>
<tr>
<td><img src="https://github.com/osmandapp/OsmAnd/assets/59626956/464a2aa5-c2d6-4fbc-8b06-6d5aa1bef092" width="300" height="auto"/></td>
</tr>
<tr>
<td><img src="https://github.com/osmandapp/OsmAnd/assets/59626956/d55a4105-5bfc-49da-b5ba-b7805dbd40f9" width="300" height="auto"/></td>
</tr>
<tr>
<td><img src="https://github.com/osmandapp/OsmAnd/assets/59626956/9ad6bbc4-bf7d-4f45-85b1-b499671bf2f4" width="300" height="auto"/></td>
</tr>
<tr>
<td><img src="https://github.com/osmandapp/OsmAnd/assets/59626956/45ef70b0-384b-4086-b167-9fc66938634a" width="300" height="auto"/></td>
</tr>
</table>

**Comment by DmitryAlexei** [comment]
@bobeeeze
this issue is considered as *Not planned*:
- The next turn widget each time displays the correct exit to be used 
- OsmAnd do not use preloaded images for this type of widgets, the picture of a roundabout is recreated every time according to map data
- as stated above, *calculation of a turn angle is a faulty operation*

**Comment by vmicho** [comment]
I would fix the geometry of the roundabout. The splitted entry/exit segments have a very pronounced "V" shepe that spoils the entry/exit angles and therefore a "going straight thru roundabout" will be rendered more like left turn -> I would make them more perpendicular to the roundabout, a bit more "U" shaped.
This does not happen when you have a simple 2way entry/exit, the minimap is always correct.

**Comment by vmicho** [comment]
here is an example of modification, this renders the mini roundabout preview more correctly: 

![Image](https://github.com/user-attachments/assets/da6c8e84-f4ee-4a24-8564-aac2e60c168b)

**Comment by Max1234-Ita** [comment]
I disagree, mapping should not be made fir rendering, as it might look fine in osmand and very bad in other apps.

The road geometry should correspond as much as possible to the reality, it's the app's task to render and ise it properly.

**Comment by vmicho** [comment]
Well if the "V" shape is very exaggerated, it is not copying reality much anyway. Also there is nothing wrong mapping for logic (like adding connecting path when one road has separately mapped footway and the other road has not, to be able to navigate correctly ...)
